### PR TITLE
insights: document aggregations feature and limitations

### LINKED ISSUE #40805: Fill out docs page for search aggregations
The page should explain the feature and the known limitations of proactive (and non-proactive) aggregation insights.

The "Learn more" link in the UI needs to be updated to point to the page.

/cc @joelkw @felixfbecker @vovakulikov @unclejustin

### PR DESCRIPTION
**Pull request by leonore**
closes #40805

stacked on https://github.com/sourcegraph/sourcegraph/pull/41160

## Test plan

`sg run docsite` for the docs

Run `sg start enteprise-codeinsights` and make sure learn more link works

## App preview:

- [Web](https://sg-web-leo-insights-4-0-docs.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-mzyncttcld.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: doc/code_insights/explanations/search_results_aggregations.md [hunk 1]
```diff
@@ -25,24 +25,57 @@ You can turn off just the proactive aggregations with the feature flag: `disable
...
 
 ### Saving aggregations to a code insights dashboard
 
 Saving aggregations to a dashboard of code insights is not yet available. 
 
-### Other (limitations perhaps for only certain search types)
+### Slow diff and commit queries
+
+Running aggregations by author is only allowed for type:diff and type:commit queries, which are likely not to complete 
+within a 2-second timeout. You can trigger an explicit search with no timeout, or you can refine your query using a
```
Comment (chwarwick) [inline doc/code_insights/explanations/search_results_aggregations.md:75]:
```suggestion
within a 2-second timeout. You can trigger an explicit search with an extended timeout, or you can refine your query using a 
```

Comment (chwarwick) [reply doc/code_insights/explanations/search_results_aggregations.md:75]: it's currently 1 min

**Comment by felixfbecker** [comment]
@leonore small nit for the future: There are a lot of odd mid-sentence hard line wraps in here, which make reviewing a bit weird. My recommendation for markdown wrapping is to have one line per sentence (hard wrap only after every full stop), that makes merging/merge conflicts and providing suggestions and comments in GitHub PR reviews the easiest.

image

**Comment by leonore** [comment]
> @leonore small nit for the future: There are a lot of odd mid-sentence hard line wraps in here, which make reviewing a bit weird. My recommendation for markdown wrapping is to have one line per sentence

my bad, leftover from an old job :) will fix

File: doc/code_insights/explanations/search_results_aggregations.md [hunk 2]
```diff
@@ -25,24 +25,63 @@ You can turn off just the proactive aggregations with the feature flag: `disable
...
 
-### Other (limitations perhaps for only certain search types)
+### Slow diff and commit queries
+
+Running aggregations by author is only allowed for type:diff and type:commit queries, which are likely not to complete within a 2-second timeout.
+You can trigger an explicit search with an extended 1-minute timeout, or you can refine your query using a single-repo filter (like `repo:^github\.com/sourcegraph/sourcegraph$`) in order to bypass this.
+
+### Structural searches
+
+Structural searches are slow and aggregations are unlikely to complete within a 2-second timeout for such searches. You can try to trigger an explicit aggregation for such cases.
```
Comment (chwarwick) [inline doc/code_insights/explanations/search_results_aggregations.md:80]:
```suggestion
Aggregations for structural searches unlikely to complete within a 2-second timeout for such searches. You can try to trigger an explicit aggregation for such cases.
```
Trying to soften that language some.

**Review by chwarwick — APPROVED** [review]

**Pull request merged**